client/subscriptions.go

```diff
@@ -0,0 +1,137 @@
+package client
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/EventStore/EventStore-Client-Go/internal/protoutils"
+	"github.com/EventStore/EventStore-Client-Go/position"
+	system_metadata "github.com/EventStore/EventStore-Client-Go/systemmetadata"
+
+	"github.com/EventStore/EventStore-Client-Go/messages"
+	api "github.com/EventStore/EventStore-Client-Go/protos/streams"
+)
+
+type SubscriptionEvent struct {
+	EventAppeared     *messages.RecordedEvent
+	Dropped           *SubscriptionDropped
+	CheckPointReached *position.Position
+}
+
+type SubscriptionDropped struct {
+	Error error
+}
+
+type Subscription struct {
+	client  *Client
+	id      string
+	inner   api.Streams_ReadClient
+	channel chan (chan *SubscriptionEvent)
+}
+
+func NewSubscription(client *Client, inner api.Streams_ReadClient, id string) *Subscription {
+	channel := make(chan (chan *SubscriptionEvent))
+
+	// It is not safe to consume a stream in different goroutines. This is why we only consume
```

Comment:
Is it the idea that a subscription can be used by multiple go routines ?
Would think it should be 1 goroutine per subscription 🤔

Reply:
Go gRPC docs say that sharing a read-stream among different goroutines is unsafe. It's safe for a write-stream though